pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1595,41 +1662,50 @@ public CompletableFuture<Void> close(
             }
         }
 
-        CompletableFuture<Void> clientCloseFuture = closeWithoutWaitingClientDisconnect
-                ? CompletableFuture.completedFuture(null)
-                : FutureUtil.waitForAll(futures);
-
-        clientCloseFuture.thenRun(() -> {
-            // After having disconnected all producers/consumers, close the managed ledger
+        CompletableFuture<Void> disconnectClientsFuture = FutureUtil.waitForAll(futures);
```

Comment:
I think `waitForAll` has overhead like the below. Plz do not call this if closeWithoutWaitingClientDisconnect = false.

https://github.com/frohoff/jdk8u-jdk/blob/master/src/share/classes/java/util/concurrent/CompletableFuture.java#L1294-L1314

Reply:
Since we need to initialize the variable `closeFutures.waitDisconnectClients`, this feature is needed 😂